CRM/Contact/Page/View/Summary.php

```diff
@@ -242,7 +234,11 @@ public function view() {
     // rather than adding to defaults for transparency - this is some old
     // copy & paste.
     $this->assign($defaults);
-    $this->assign('allTabs', $this->getTabs($defaults));
+    $this->assign('allTabs', $this->getTabs([
+      'contact_type' => $defaults['contact_type'],
+      'contact_sub_type' => $defaults['contact_sub_type'],
```

Comment:
I'm getting an error about undefined contact_sub_type here (the site has no such things).

Otherwise though looks much better.

Reply:
Ah yes - cos that function helpfully filters out empty fields - I've switched to our more recent apiv4 based methodology - it does do an extra look up but we can use it remove some others